exchanges/kraken/kraken_test.go

```diff
@@ -1816,150 +1817,120 @@ func TestWsOwnTrades(t *testing.T) {
-        "userref": 0,
-        "vol": "0.00000010",
-        "vol_exec": "0.00000000"
-      }
-    },
-    {
-      "OGTT3Y-C6I3P-XRI6HX": {
-        "cost": "0.00000",
-        "descr": {
-          "close": "",
-          "leverage": "0.1",
-          "order": "sell 0.00001000 XBT/USD @ limit 90.40000 with 0:1 leverage",
-          "ordertype": "limit",
-          "pair": "XBT/USD",
-          "price": "90.40000",
-          "price2": "0.00000",
-          "type": "sell"
-        },
-        "expiretm": "0.000000",
-        "fee": "0.00000",
-        "limitprice": "90.40000",
-        "misc": "",
-        "oflags": "fcib",
-        "opentm": "0.000000",
-        "price": "90.40000",
-        "refid": "OKIVMP-5GVZN-Z2D2UA",
-        "starttm": "0.000000",
-        "status": "open",
-        "stopprice": "0.000000",
-        "userref": 0,
-        "vol": "0.00001000",
-        "vol_exec": "0.00000000"
-      }
-    },
-    {
-      "OGTT3Y-C6I3P-XRI6HX": {
-        "cost": "0.00000",
-        "descr": {
-          "close": "",
-          "leverage": "0.1",
-          "order": "sell 0.00001000 XBT/USD @ limit 9.00000 with 0:1 leverage",
-          "ordertype": "limit",
-          "pair": "XBT/USD",
-          "price": "9.00000",
-          "price2": "0.00000",
-          "type": "sell"
-        },
-        "expiretm": "0.000000",
-        "fee": "0.00000",
-        "limitprice": "9.00000",
-        "misc": "",
-        "oflags": "fcib",
-        "opentm": "0.000000",
-        "price": "9.00000",
-        "refid": "OKIVMP-5GVZN-Z2D2UA",
-        "starttm": "0.000000",
-        "status": "open",
-        "stopprice": "0.000000",
-        "userref": 0,
-        "vol": "0.00001000",
-        "vol_exec": "0.00000000"
-      }
-    }
-  ],
-  "openOrders"
-]`)
-	err := k.wsHandleData(pressXToJSON)
-	if err != nil {
-		t.Error(err)
+	pairs := currency.Pairs{
+		currency.Pair{Base: currency.XBT, Quote: currency.USD},
+		currency.Pair{Base: currency.XBT, Quote: currency.USDT},
+	}
+	k := Kraken{
+		Base: exchange.Base{
+			Name: "dummy",
+			CurrencyPairs: currency.PairsManager{
+				Pairs: map[asset.Item]*currency.PairStore{
+					asset.Spot: {
+						Available: pairs,
+						Enabled:   pairs,
+						ConfigFormat: &currency.PairFormat{
+							Uppercase: true,
+							Delimiter: currency.DashDelimiter,
+						},
+					},
+				},
+			},
+			Websocket: &stream.Websocket{
+				Wg:          new(sync.WaitGroup),
+				DataHandler: make(chan interface{}, 128),
+			},
+		},
 	}
-	pressXToJSON = []byte(`[
-  [
-    {
-      "OGTT3Y-C6I3P-XRI6HX": {
-        "status": "closed"
-      }
-    },
-    {
-      "OGTT3Y-C6I3P-XRI6HX": {
-        "status": "closed"
-      }
-    }
-  ],
-  "openOrders"
-]`)
-	err = k.wsHandleData(pressXToJSON)
+
+	k.API.Endpoints = k.NewEndpoints()
+
+	fixture, err := os.Open("testdata/wsOpenTrades.json")
 	if err != nil {
+		t.Errorf("Error opening test fixture 'testdata/wsOpenTrades.json': %v", err)
+		return
+	}
```

Comment:
Can add a `defer fixture.Close()` here or `fixture.Close()` at the end of the test to explicitly close the file handle

Reply:
Can do.
Rationale: It's a test, and fixture is lexically scoped.
Therefore it gets closed when fixture is GC'd as part of the finalizers.
Doesn't hurt to defer it though if you prefer.

Reply:
Oh, yuk!
Linter requires checking the return value!
Think I'll switch to testify during this PR then and use assert.Nil

Reply:
@thrasher- Fixed with [e557e85](https://github.com/thrasher-corp/gocryptotrader/pull/1327/commits/e557e85fcd25b97535a1e14b2600ccc2da7e188d)